esp-hal/src/timer/systimer.rs

```diff
@@ -169,51 +107,31 @@ impl<'d> SystemTimer<'d> {
         // worst case scenario the second accessor ends up reading
         // an older time stamp
 
-        let unit = unsafe { SpecificUnit::<'_, 0>::conjure() };
-        unit.read_count()
+        Unit::Unit0.read_count()
     }
 }
 
-impl SystemTimer<'static> {
-    /// Split the System Timer into three alarms.
-    ///
-    /// This is a convenience method to create `'static` alarms of the same
-    /// type. You are encouraged to use [Alarm::new] over this very specific
-    /// helper.
-    pub fn split<MODE>(self) -> SysTimerAlarms<MODE, Blocking> {
-        static mut UNIT0: Option<AnyUnit<'static>> = None;
-        let unit0 = unsafe { &mut *addr_of_mut!(UNIT0) };
-
-        let unit0 = unit0.insert(self.unit0.into());
-        let unit = FrozenUnit::new(unit0);
-
-        SysTimerAlarms {
-            alarm0: Alarm::new(self.comparator0.into(), &unit),
-            alarm1: Alarm::new(self.comparator1.into(), &unit),
-            alarm2: Alarm::new(self.comparator2.into(), &unit),
-            #[cfg(not(esp32s2))]
-            unit1: self.unit1,
-        }
-    }
+/// Alarms created from the System Timer peripheral.
+pub struct SysTimerAlarms {
```

Comment:
```suggestion
pub struct SystemTimerAlarms {
```

We should probably be consistent with our naming.